[RIP-21] Setup JPA

MrZhenShen opened the pull request:
# Description
## Spring Data JPA

[Task {RIP-21}](https://sigma-internship-project.atlassian.net/browse/RIP-21?atlOrigin=eyJpIjoiOTg2OGI4YjM4Y2YxNDdkNWJlYzVmMzI1YjdiZmU3YTgiLCJwIjoiaiJ9)

Change type: #minor
**Create**
* Entities: User, Locale, Fund, StatisticData
* Enums: Role, Subscription
* Repositories: UserRepository, FundRepository, StatisticDataRepository

[RIP-21]: https://sigma-internship-project.atlassian.net/browse/RIP-21?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MrZhenShen committed:
create Entities and enums

MrZhenShen committed:
create repositories for each entity

eduard-romaniuk reviewed (changes requested):
Check comments and apply fixes to all places where same issues have place

MrZhenShen committed:
Delete LocaleRepository.java
Fix access modifiers in Fund entity
Add @Table annotations to each entity
Remove nullable parameter in Column annotations above ids
Use primitive data type for ids
Use Many-to-One for locale in User

MrZhenShen committed:
Use @PrePersist and @PreUpdate to feel updateDate/createDate,when entity creates or updates

MrZhenShen committed:
Add unique parameters
Remove OneToMany field in Locale
Remove fetch type for locale in User

MrZhenShen committed:
Merge branch 'main' into setup-jpa
